lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1260,6 +1276,27 @@ WitnessChecker::lookupValueWitnesses(ValueDecl *req, bool *ignoringNames) {
   assert(!isa<AssociatedTypeDecl>(req) && "Not for lookup for type witnesses*");
   
   SmallVector<ValueDecl *, 4> witnesses;
+
+  // Do an initial check to see if there are any @_implements remappings
+  // for this requirement.
+  if (req->isProtocolRequirement() && req->hasName()) {
```

Comment:
Maybe factor this out into a new lookupValueWitnessViaImplementsAttr() and use early return to avoid the "} cascade of death"